lens/sqlrepo/repo.go

```diff
@@ -1,17 +1,57 @@
 package sqlrepo
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/urfave/cli/v2"
 
+	pgchainbs "github.com/filecoin-project/go-bs-postgres-chainnotated"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/filecoin-project/sentinel-visor/lens"
 	"github.com/filecoin-project/sentinel-visor/lens/util"
 )
 
 func NewAPIOpener(c *cli.Context) (lens.APIOpener, lens.APICloser, error) {
-	bs, err := NewBlockStore(c.String("repo"))
+	pgbsCfg := pgchainbs.PgBlockstoreConfig{
+		PgxConnectString:        c.String("repo"),
+		CacheInactiveBeforeRead: true,
+		DisableBlocklinkParsing: true,
+		LogCacheStatsOnUSR1:     true,
+	}
+
+	pgbsCfg.InstanceNamespace = c.String("lens-postgres-namespace")
+	if pgbsCfg.InstanceNamespace == "" {
+		pgbsCfg.InstanceNamespace = "main"
+	}
+
+	if customCacheSize := c.Int("lens-cache-hint"); customCacheSize != 1024*1024 {
+		pgbsCfg.CacheSizeGiB = uint64(customCacheSize)
```

Comment:
In this case you need to divide here - `pgbsCfg.CacheSizeGiB` is measured in GiB

Reply:
if you specify the flag with this lens, we interpret it as in gb 😁